cyfra-core/src/main/scala/io/computenode/cyfra/core/GBufferRegion.scala

```diff
@@ -0,0 +1,61 @@
+package io.computenode.cyfra.core
+
+import io.computenode.cyfra.core.GProgram.BufferSizeSpec
+import io.computenode.cyfra.core.layout.{Layout, LayoutStruct}
+import io.computenode.cyfra.dsl.Value
+import io.computenode.cyfra.dsl.buffer.GBuffer
+
+import java.nio.ByteBuffer
+
+sealed trait GBufferRegion[ReqAlloc <: Layout: LayoutStruct, ResAlloc <: Layout: LayoutStruct]:
+  val initAlloc: ReqAlloc
+
+object GBufferRegion:
+
+  private[cyfra] case class ZeroedBuffer[T <: Value](size: Int) extends GBuffer[T]
+
+  private[cyfra] case class BufferFromRam[T <: Value](buff: ByteBuffer) extends GBuffer[T]
```

Comment:
A materialized GBuffer needs size and length:
* length - number of `T`s in a buffer
* size - number of bytes (sizeof<T> *length + padding)

Reply:
Sure, we can get that from Tag on Value.